test/unit_test/options_boost_po_test.cc

```diff
@@ -41,7 +41,7 @@ BOOST_AUTO_TEST_CASE(typed_options_parsing) {
   char command_line[] = "exe --str_opt test_str -i 5 --bool_opt yes --char_opt f --float_opt 4.3";
   int argc;
   // Only the returned char* needs to be deleted as the individual pointers simply point into command_line.
-  std::unique_ptr<char*> argv(convert_to_command_args(command_line, argc));
+  std::unique_ptr<char*[]> argv(convert_to_command_args(command_line, argc));
```

Comment:
It worries me that the type system did not catch this issue. Is there something we can do going forward to avoid cases like this?

Reply:
It didn't catch it because `convert_to_command_args` returns `char**`, so still a failure of the type system. Or rather, caused by C++'s eager implicit conversion semantics... Not using raw pointers and raw pointers to raw pointers is a good solution for avoiding things like this. Argv is a bit of a special case though

Reply:
By the way this caused a mismatched delete/delete[], it least it wasn't a leak!